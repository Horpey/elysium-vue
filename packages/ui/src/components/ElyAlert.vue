<template>
  <div class="mt-4 border border-l-8 border-koromiko-800 bg-koromiko-800/10 px-4 py-2">
    <span class="flex items-center space-x-3 font-mono text-xs text-koromiko-900">
      <span>❯</span>

      <span v-if="$slots.title" class="font-bold ">
        <slot name="title" />
      </span>
      <slot name="description" />
    </span>
  </div>
</template>
