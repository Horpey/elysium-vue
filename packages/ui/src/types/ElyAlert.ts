export const ELY_ALERT_THEME = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  TERTIARY: 'tertiary',
  DESTRUCTIVE: 'destructive',
} as const

export type ElyAlertThemeKey = keyof typeof ELY_ALERT_THEME
