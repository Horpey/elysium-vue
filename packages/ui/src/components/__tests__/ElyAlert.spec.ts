import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import ElyAlert from '../ElyAlert.vue'

describe('elyAlert', () => {
  it('renders with default props', async () => {
    const wrapper = mount(ElyAlert)

    // Assert that the component is rendered
    expect(wrapper.exists()).toBe(true)

    // Assert default values
    expect(wrapper.find('.flex.space-x-3.font-mono.text-xs').exists()).toBe(true)
    expect(wrapper.find('.flex.flex-col.space-y-1').exists()).toBe(true)

    // Assert that title and slots are not rendered
    expect(wrapper.find('.text-xs.font-bold.uppercase.tracking-wider').exists()).toBe(false)
    expect(wrapper.find('.span').exists()).toBe(false)

    // Assert snapshot
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders with custom props and slots', async () => {
    const wrapper = mount(ElyAlert, {
      props: {
        title: 'Custom Title',
      },
      slots: {
        default: '<p>Custom content</p>',
      },
    })

    // Assert that the component is rendered
    expect(wrapper.exists()).toBe(true)

    // Assert custom values
    expect(wrapper.find('.text-xs.font-bold.uppercase.tracking-wider').text()).toBe('Custom Title')
    expect(wrapper.find('p').text()).toBe('Custom content')

    // Assert snapshot
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders with custom icon', async () => {
    const wrapper = mount(ElyAlert, {
      props: {
        title: 'Custom Title',
      },
      slots: {
        icon: '<svg>mock icon</svg>',
      },
    })

    // Assert that the component is rendered
    expect(wrapper.exists()).toBe(true)

    // Assert that the icon slot is rendered
    expect(wrapper.find('svg').exists()).toBe(true)

    // Assert snapshot
    expect(wrapper.html()).toMatchSnapshot()
  })
})
