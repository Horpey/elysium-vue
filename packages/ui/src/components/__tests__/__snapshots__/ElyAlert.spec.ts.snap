// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`elyAlert > renders with custom icon 1`] = `
"<div class=\\"clip-top-right relative border border-l-8 px-3 py-2 border-koromiko-700 bg-koromiko-800/10 text-koromiko-900\\"><span class=\\"flex space-x-3 font-mono text-xs\\"><span class=\\"h-4 w-4\\"><svg>mock icon</svg></span>
  <div class=\\"flex flex-col space-y-1\\"><span class=\\"text-xs font-bold uppercase tracking-wider\\">Custom Title</span><span></span></div></span>
</div>"
`;

exports[`elyAlert > renders with custom props and slots 1`] = `"<div class=\\"clip-top-right relative border border-l-8 px-3 py-2 border-koromiko-700 bg-koromiko-800/10 text-koromiko-900\\"><span class=\\"flex space-x-3 font-mono text-xs\\"><!--v-if--><div class=\\"flex flex-col space-y-1\\"><span class=\\"text-xs font-bold uppercase tracking-wider\\">Custom Title</span><span><p>Custom content</p></span></div></span></div>"`;

exports[`elyAlert > renders with default props 1`] = `"<div class=\\"clip-top-right relative border border-l-8 px-3 py-2 border-koromiko-700 bg-koromiko-800/10 text-koromiko-900\\"><span class=\\"flex space-x-3 font-mono text-xs\\"><!--v-if--><div class=\\"flex flex-col space-y-1\\"><!--v-if--><span></span></div></span></div>"`;
